Installation issue: gdrcopy

When I try to install gdrcopy it builds but tries to install in /usr/local/lib as opposed to my spack sandbox.  I modified gdrcopy/package.py to fix the problem.

I get the following error:

==> Installing gdrcopy-master-gn4oaromvgzkq72ld66jpynzutojejw6
==> No binary for gdrcopy-master-gn4oaromvgzkq72ld66jpynzutojejw6 found: installing from source
==> gdrcopy: Executing phase: 'edit'
==> gdrcopy: Executing phase: 'build'
==> gdrcopy: Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'lib_install' 'PREFIX=/home/mjleven/local/spatse/install/spack/opt/spack/linux-rhel8-thunderx2/gcc-10.2.0/gdrcopy-master-gn4oaromvgzkq72ld66jpynzutojejw6'
 
3 errors found in build log:
     18    make LIB_MAJOR_VER=2 LIB_MINOR_VER=2
     19    make[1]: Entering directory '/tmp/mjleven/spack-stage/spack-stage-gdrcopy-master-gn4oaromvgzkq72ld66jpynzutojejw6/spack-src/src'
     20    GDRAPI_ARCH=ARM64
     21    make[1]: Leaving directory '/tmp/mjleven/spack-stage/spack-stage-gdrcopy-master-gn4oaromvgzkq72ld66jpynzutojejw6/spack-src/src'
     22    installing in /usr/local/lib /usr/local/include...
     23    'src/libgdrapi.so.2.2' -> '/usr/local/lib/libgdrapi.so.2.2'
  >> 24    install: cannot create regular file '/usr/local/lib/libgdrapi.so.2.2': Permission denied
  >> 25    ln: failed to create symbolic link 'libgdrapi.so.2': Permission denied
  >> 26    ln: failed to create symbolic link 'libgdrapi.so': Permission denied
     27    make: *** [Makefile:68: lib_install] Error 1

I fixed the problem with a change to gdrcopy/package.py:

[mjleven@stnv1 spack]$ git diff
diff --git a/var/spack/repos/builtin/packages/gdrcopy/package.py b/var/spack/repos/builtin/packages/gdrcopy/package.py
index 88cf814e40..14ba4bb31b 100644
--- a/var/spack/repos/builtin/packages/gdrcopy/package.py
+++ b/var/spack/repos/builtin/packages/gdrcopy/package.py
@@ -32,4 +32,4 @@ def build(self, spec, prefix):
     def install(self, spec, prefix):
         mkdir(prefix.include)
         mkdir(prefix.lib64)
-        make('lib_install', 'PREFIX={0}'.format(self.prefix))
+        make('lib_install', 'prefix={0}'.format(self.prefix))
[mjleven@stnv1 spack]$

--- a/var/spack/repos/builtin/packages/gdrcopy/package.py
+++ b/var/spack/repos/builtin/packages/gdrcopy/package.py
@@ -26,4 +26,9 @@ class Gdrcopy(MakefilePackage):
     def install(self, spec, prefix):
         mkdir(prefix.include)
         mkdir(prefix.lib64)
-        make('lib_install', 'PREFIX={0}'.format(self.prefix))
+        if spec.satisfies('@2.2:'):
+            make('lib_install',
+                 'prefix={0}'.format(self.prefix))
+        else:
+            make('lib_install',
+                 'PREFIX={0}'.format(self.prefix))

